test/tools/cli/IonProcess.java

```diff
@@ -61,66 +62,68 @@ public static void main(final String[] args) {
         }
     }
 
     private static void processFiles(IonWriter ionWriterForOutput,
                                      IonWriter ionWriterForErrorReport,
-                                     ProcessArgs args) throws IOException {
+                                     ProcessArgs args,
+                                     OutputStream outputStream) throws IOException {
         if (args.getOutputFormat() == OutputFormat.EVENTS) {
             ionWriterForOutput.writeSymbol(EVENT_STREAM);
         }
         //this object stores data for ErrorReport.
-        CurrentInfo currentInfo = new CurrentInfo(null, 1, null);
+        ProcessContext processContext = new ProcessContext(null, 1, null, null);
 
         for (String path : args.getInputFiles()) {
             try (
                     InputStream inputStream = new BufferedInputStream(new FileInputStream(path));
                     IonReader ionReader = IonReaderBuilder.standard().build(inputStream);
-                    ) {
-
-                currentInfo.setFileName(path);
+            ) {
+                processContext.setFileName(path);
 
                 if (isEventStream(ionReader)) {
-                    currentInfo.setEventIndex(currentInfo.getEventIndex() - 1);
-                    processFromEventStream(ionWriterForOutput, ionWriterForErrorReport, ionReader, currentInfo, args);
+                    processContext.setEventIndex(processContext.getEventIndex() - 1);
+                    processFromEventStream(ionWriterForOutput, ionWriterForErrorReport,
+                            ionReader, processContext, args, outputStream);
                 } else {
-                    processFromIonStream(ionWriterForOutput, ionWriterForErrorReport, ionReader, currentInfo, args);
+                    processFromIonStream(ionWriterForOutput, ionWriterForErrorReport, ionReader, processContext, args);
                 }
 
                 ionWriterForOutput.finish();
                 ionWriterForErrorReport.finish();
             } catch (IonException e) {
-                if (args.getOutputFormat() == OutputFormat.EVENTS) {
-                    new ErrorDescription(ErrorType.READ, e.getMessage(), currentInfo.getFileName(),
-                            currentInfo.getEventIndex()).writeOutput(ionWriterForErrorReport);
+                if (args.getOutputFormat() == OutputFormat.EVENTS || processContext.getState() == ErrorType.WRITE) {
+                    new ErrorDescription(processContext.getState(), e.getMessage(), processContext.getFileName(),
+                            processContext.getEventIndex()).writeOutput(ionWriterForErrorReport);
                 } else {
-                    new ErrorDescription(ErrorType.READ, e.getMessage(), currentInfo.getFileName())
+                    new ErrorDescription(processContext.getState(), e.getMessage(), processContext.getFileName())
                             .writeOutput(ionWriterForErrorReport);
```

Comment:
Looks like these two branches only differ by whether the event index is provided. Can you unify the branches and handle the presence/absence of an event index within a single ErrorDescription constructor?

Reply:
Changed.

used initial value -1. (IonProcess.java 67). the writeOutput method will ignore eventIndex if it's -1.